internal/runtimehandlerhooks/high_performance_hooks.go

```diff
@@ -259,50 +217,25 @@ func setCPUSLoadBalancing(c *oci.Container, enable bool, schedDomainDir string)
 		return fmt.Errorf("find container %s CPUs", c.ID())
 	}
 
-	cpus, err := cpuset.Parse(lspec.Resources.CPU.Cpus)
-	if err != nil {
-		return err
+	if node.CgroupIsV2() {
+		return fmt.Errorf("unsupported")
 	}
 
-	for _, cpu := range cpus.ToSlice() {
-		cpuSchedDomainDir := fmt.Sprintf("%s/cpu%d", schedDomainDir, cpu)
-		err := filepath.Walk(cpuSchedDomainDir, func(path string, info os.FileInfo, err error) error {
-			if err != nil {
-				return err
-			}
-			if !info.Mode().IsRegular() || info.Name() != "flags" {
-				return nil
-			}
-			content, err := os.ReadFile(path)
-			if err != nil {
-				return err
-			}
-
-			flags, err := strconv.Atoi(strings.Trim(string(content), "\n"))
-			if err != nil {
-				return err
-			}
-
-			var newContent string
-			if enable {
-				newContent = strconv.Itoa(flags | 1)
-			} else {
-				// we should set the LSB to 0 to disable the load balancing for the specified CPU
-				// in case of sched domain all flags can be represented by the binary number 111111111111111 that equals
-				// to 32767 in the decimal form
-				// see https://github.com/torvalds/linux/blob/0fe5f9ca223573167c4c4156903d751d2c8e160e/include/linux/sched/topology.h#L14
-				// for more information regarding the sched domain flags
-				newContent = strconv.Itoa(flags & 32766)
-			}
+	pid, err := c.Pid()
+	if err != nil {
+		return fmt.Errorf("failed to get pid of container %s: %w", c.ID(), err)
+	}
+	controllers, err := cgroups.ParseCgroupFile("/proc/" + strconv.Itoa(pid) + "/cgroup")
+	if err != nil {
+		return fmt.Errorf("failed to get cgroups of container %s: %w", c.ID(), err)
+	}
 
-			return os.WriteFile(path, []byte(newContent), 0o644)
-		})
-		if err != nil {
-			return err
-		}
+	cpusetPath, ok := controllers["cpuset"]
+	if !ok {
+		return fmt.Errorf("failed to get cpuset of container %s", c.ID())
 	}
 
-	return nil
+	return cgroups.WriteFile("/sys/fs/cgroup/cpuset"+cpusetPath, "cpuset.sched_load_balance", "0")
```

Comment:
This needs much better docs right here explaining that all other cpusets must lose access to the cpus or this does not work. Where is that part going to be handled? Just by setting up the slices properly?

Reply:
added documentation to the top of the function!